### PR TITLE
Change auth to use refresh_token

### PR DESCRIPTION
**Pull request by nelsongraca**
Since I was having trouble with configuring the script to keep the car awake with username/password I have adjusted everything to authenticate only with the refresh_token.

### CODE REVIEW

File: run/tesla_api.py [hunk 1]
```diff
@@ -143,16 +144,14 @@ def _get_api_token():
...
 
-    # If the access token is not already stored in tesla_api_json AND the
-    # user didn't provide a token pair in the teslausb_setup_variables.conf,
-    # attempt to use the login credentials from teslausb_setup_variables.conf
-    # to create a new token pair, if they exist.
-    elif SETTINGS['tesla_email'] and SETTINGS['tesla_password']:
-        # Create a new pair of tokens from the OAuth API.
-        tesla = teslapy.Tesla(SETTINGS['tesla_email'], SETTINGS['tesla_password'])
-        tesla.fetch_token()
+    # If the access token is not already stored in tesla_api_json AND the user provided a refresh_token force it into the client to get a proper token
```
Comment (marcone) [inline run/tesla_api.py]: break this up into two lines

File: run/tesla_api.py [hunk 2]
```diff
@@ -258,92 +257,14 @@ def list_vehicles():
     return _execute_request(base_url, None, None, False)
 
 
-def get_service_data():
```
Comment (marcone) [inline run/tesla_api.py]: why did you delete all these?

File: setup/pi/setup-teslausb [hunk 3]
```diff
@@ -27,8 +27,7 @@ function read_setup_variables {
   newnamefor[musicsharename]=MUSIC_SHARE_NAME
   newnamefor[shareuser]=SHARE_USER
   newnamefor[sharepassword]=SHARE_PASSWORD
-  newnamefor[tesla_email]=TESLA_EMAIL
-  newnamefor[tesla_password]=TESLA_PASSWORD
+  newnamefor[tesla_refresh_token]=TESLA_REFRESH_TOKEN
```
Comment (marcone) [inline setup/pi/setup-teslausb]: This section is for providing backwards compatibility with old config files. Since this is a new setup variable, this isn't needed here.

**Review by marcone — COMMENTED** [review]
I know this is a WIP, but I added some preliminary comments

**Comment by nelsongraca** [comment]
@marcone please feel free to add comments even though it's a WIP that's why I already created the PR to have your feedback earlier

File: run/tesla_api.py (hunk 2 above)
Comment (nelsongraca) [reply run/tesla_api.py]: Seems like unused code and not relevant for the job of keeping the car awake so I removed it.
Can put it back if it's a blocker.

File: run/tesla_api.py (hunk 1 above)
Comment (nelsongraca) [reply run/tesla_api.py]: what's the maximum line length?

File: run/tesla_api.py (hunk 2 above)
Comment (marcone) [reply run/tesla_api.py]: I would prefer to keep these.

File: run/tesla_api.py (hunk 1 above)
Comment (marcone) [reply run/tesla_api.py]: Please keep it under 100 characters for readability on smaller screens

Comment (nelsongraca) [reply run/tesla_api.py]: done

**Comment by miles267** [comment]
@nelsongraca once this has been implemented, will it restore the functionality that kept the car awake during a sync provided 2FA has not been enabled?  Thanks.

**Comment by nelsongraca** [comment]
@miles267 This will even work with 2FA enabled, since it uses an API refresh_token, same thing as TeslaMate for example.
As for keeping the car awake I did not change the methods that can be used so should work

**Comment by nelsongraca** [comment]
@marcone this is ready for review.

**Comment by miles267** [comment]
Awesome. This is exciting. Once this is committed and available in the main-dev branch can’t wait to use it. May also then start using the music volume too.
________________________________

**Comment by marcone** [comment]
> @marcone this is ready for review.

Can you put back the functions you removed?

**Comment by nelsongraca** [comment]
> > @marcone this is ready for review.
> 
> Can you put back the functions you removed?

Sorry thought I had put them all back will update

**Comment by nelsongraca** [comment]
should be good now
